pymc_marketing/product_incrementality/mv_its.py

```diff
@@ -0,0 +1,403 @@
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Multivariate Interrupted Time Series Analysis for Product Incrementality."""
+
+import arviz as az
+import matplotlib.pyplot as plt
+import matplotlib.ticker as mtick
+import numpy as np
+import pandas as pd
+import pymc as pm
+
+HDI_ALPHA = 0.5
+
+
+class MVITS:
+    """Class to perform a multivariate interrupted time series analysis with the
+    specific intent of determining where the sales of a new product came from."""
+
+    def __init__(
+        self,
+        data: pd.DataFrame,
+        treatment_time,
+        background_sales: list[str],
+        innovation_sales: str,
+        market_saturated: bool = True,
+        rng=42,
+        sample_kwargs: dict | None = None,
+    ):
+        self.data = data
+        self.treatment_time = treatment_time
+        self.background_sales = background_sales
+        self.innovation_sales = innovation_sales
+        self.rng = rng
+        self.sample_kwargs = sample_kwargs if sample_kwargs is not None else {}
+        self.market_saturated = market_saturated
+
+        # build the model
+        self.model = self.build_model(
+            self.data[self.background_sales],
+            self.data[self.innovation_sales],
+            self.market_saturated,
+            treatment_time=self.treatment_time,
+        )
+
+        # sample from prior, posterior, posterior predictive
+        with self.model:
+            self.idata = pm.sample_prior_predictive(random_seed=self.rng)
+            self.idata.extend(pm.sample(**self.sample_kwargs, random_seed=self.rng))
+            self.idata.extend(
+                pm.sample_posterior_predictive(
+                    self.idata,
+                    var_names=["mu", "y"],
+                    random_seed=self.rng,
+                )
+            )
+
+        # Calculate the counterfactual background sales, if the new product had not been introduced
+        zero_sales = np.zeros(self.data[self.innovation_sales].shape, dtype=np.int32)
+        self.counterfactual_model = pm.do(self.model, {"innovation_sales": zero_sales})
+        with self.counterfactual_model:
+            self.idata_counterfactual = pm.sample_posterior_predictive(
+                self.idata, var_names=["mu", "y"], random_seed=self.rng
+            )
+
+        return
+
+    @staticmethod
+    def build_model(
+        background_sales: pd.DataFrame,
+        innovation_sales: pd.Series,
+        market_saturated: bool,
+        treatment_time,
+        *,
+        alpha_background=0.5,
+    ):
+        """Return a PyMC model for a multivariate interrupted time series analysis."""
+
+        if not background_sales.index.equals(innovation_sales.index):
+            raise ValueError(
```

Comment:
Do you want to add a test for this? If not, just add a no-cover statement so codecov doesn’t complain